Modules/IO/GDCM/test/itkGDCMImageIOIntToUintTest.cxx

```diff
@@ -0,0 +1,77 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkGDCMImageIO.h"
+#include "itkImageFileReader.h"
+#include "itkTestingMacros.h"
+
+// Specific ImageIO test
+
+/** This test verifies itkGDCMImageIO can read
+ *  DICOM files that contains Int data that end up rescaled into Uint
+ */
+int
+itkGDCMImageIOIntToUintTest(int argc, char * argv[])
+{
+  if (argc < 2)
+  {
+    std::cerr << "Missing parameters." << std::endl;
+    std::cerr << "Usage: " << itkNameOfTestExecutableMacro(argv);
+    std::cerr << " DicomImage" << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  using InputPixelType = unsigned short;
+  using InputImageType = itk::Image<InputPixelType, 2>;
+  using ReaderType = itk::ImageFileReader<InputImageType>;
+  using ImageIOType = itk::GDCMImageIO;
+
+  auto       dcmImageIO = ImageIOType::New();
+  const bool canRead = dcmImageIO->CanReadFile(argv[1]);
+  if (!canRead)
+  {
+    std::cerr << "Cannot read file " << argv[1] << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  auto reader = ReaderType::New();
+  reader->SetFileName(argv[1]);
+  reader->SetImageIO(dcmImageIO);
+
+  try
```

Comment:
This try-catch block could (and should) be replaced by `ITK_TRY_EXPECT_NO_EXCEPTION(reader->Update());`